packages/ui/libs/user/logic/src/admin/trades.ts

```diff
@@ -0,0 +1,72 @@
+import { TableComponentProps, TableRowData } from '@energyweb/origin-ui-core';
+import { TradeForAdminDTO } from '@energyweb/exchange-irec-react-query-client';
+import { useTranslation } from 'react-i18next';
+import { formatDate, EnergyFormatter } from '@energyweb/origin-ui-utils';
+
+type TUseTradesTableLogicArgs = {
+  trades: TradeForAdminDTO[];
+  loading: boolean;
+};
+
+type TUseTradesTableLogic = (
+  args: TUseTradesTableLogicArgs
+) => TableComponentProps<string>;
+
+type TFormatTrades = (
+  args: Omit<TUseTradesTableLogicArgs, 'loading'>
+) => TableRowData<string>[];
+
+type FormattedTrades = {
+  id: string;
+  dateAndTime: string;
+  buyerId: string;
+  sellerId: string;
+  volume: string;
+  certificateId: string;
+  price: number;
+  total: string;
+};
+
+const formatTrades: TFormatTrades = ({ trades }): FormattedTrades[] => {
+  return (
+    trades?.map((trade) => {
+      const price = trade.price / 100;
+      const totalPrice =
+        parseFloat(EnergyFormatter.format(trade.volume)) * price;
```

Comment:
I'm not sure what numbers are there exactly, but to minimize losing precision it would be better to multiply it first and then divide by `100`. Also I'm not sure what `100` is? Is that `centsToUSD` ratio?

Reply:
it's an implementation from other pages so I didn't change it